addons/point_of_sale/static/src/app/store/db.js

```diff
@@ -554,11 +554,36 @@ export class PosDB {
         }
         return list;
     }
+
+    get_product_by_template_id(tmpl_id) {
+        let list = [];
+        const products = Object.values(this.product_by_id);  // Récupère un tableau des valeurs de this.product_by_id
+        for (let product of products) {  // Itère sur les produits dans le tableau
+            if (product.product_tmpl_id === tmpl_id) {
+                list.push(product);
+            }
+        }
+        return list;
+    }
+    
     /* returns a list of products with :
      * - a category that is or is a child of category_id,
      * - a name, package or barcode containing the query (case insensitive)
      */
     search_product_in_category(category_id, query) {
+        let product_tmpl_id;
+        let filteredProducts = [];
+        try {
+            let reg = /product\.tmpl\.id:(\d+)/gi;
+            let match = reg.exec(query);
+            if (match) {
+                product_tmpl_id = parseInt(match[1], 10);
+                filteredProducts = this.get_product_by_template_id(product_tmpl_id);
+                query = query.replace(/;product\.tmpl\.id:\d+/gi, '');
+            }
+        } catch {
+            // Log or handle error if necessary
```

Comment:
Why do you do nothing ? :thinking:

Reply:
i just want to try i dont really care about the catch.

Reply:
you should at least put a `console.error()` with the catched error in my opinion, except if we know that it will fail on purpose. It's very bad practice to ignore relevant errors